app/javascript/src/components/GameLog.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="border border-black rounded p-2 m-2">
+    
+    <div>
+      <GameLogEntry
+        v-for="(entry, index) in processedLog()"
+          :events="entry"
+          :key="index"
+      ></GameLogEntry>
+    </div>
+  </div>
+</template>
+
+<script>
+import GameLogEntry from "./GameLogEntry.vue";
+export default {
+  name: "GameLog",
+  props: { log: Array },
+  components: { GameLogEntry },
+  methods: {
+    processedLog() {
+      let rawlog = this.log || [];
+      let entries = [];
+      rawlog.forEach((action) => {
+        if (action.type === "initialize") {
+          entries.push({event: action});
+        } else if (action.type === "rondel") {
+          entries.push({event: [action]});
+        }
+        else {
+          entries[entries.length - 1].event.push(action);
+        }
+      });
+      return entries.slice().reverse();
```

Comment:
Very small point but I don't _think_ you need `slice()` here. `reverse()` does indeed mutate the array but since we re-generate the array each time this is called, I think that it would be harmless to make the mutation here. Does it work if we remove `slice()`?

Reply:
I did some tests and it looks like we can remove `slice()`.